ntoskrnl/mm/balance.c

```diff
@@ -39,6 +39,18 @@ static LONG PageOutThreadActive;
 
 /* FUNCTIONS ****************************************************************/
 
+static inline VOID UpdatePeakCommitment(VOID)
+{
+    // HACK: MmTotalCommittedPages should be adjusted consistently with
+    // other counters at different places.
+    MmTotalCommittedPages = MiMemoryConsumers[MC_SYSTEM].PagesUsed +
+                          MiMemoryConsumers[MC_USER].PagesUsed +
+                          MiUsedSwapPages;
+
+    if (MmTotalCommittedPages > MmPeakCommitment)
+        MmPeakCommitment = MmTotalCommittedPages;
```

Comment:
Related to the locking question above... atomically updating a maximum value looks something like this:
```
SIZE_T TotalCommittedPages = ...; // obtain this value atomically above instead of re-reading from memory
SIZE_T PeakCommitment = MmPeakCommitment;
while (TotalCommittedPages > PeakCommitment &&
       InterlockedCompareExchangeSizeT(&MmPeakCommitment, TotalCommittedPages, PeakCommitment) != PeakCommitment)
{
    PeakCommitment = MmPeakCommitment;
}
```

Reply:
See previous comment / question regarding lock